tests/test.py

```diff
@@ -0,0 +1,252 @@
+from unittest import TestCase
+import solvespace as slvs
+from math import radians
+
+class CoreTest(TestCase):
+  def test_crank_rocker(self):
+    """Crank rocker example."""
+    print("Crank rocker")
+    slvs.clear()
+    g = 1
+    wp = slvs.add_2d_base(g)
+    p0 = slvs.add_point_2d(g, 0, 0, wp)
+    slvs.dragged(g, p0, wp)
+    p1 = slvs.add_point_2d(g, 90, 0, wp)
+    slvs.dragged(g, p1, wp)
+    line0 = slvs.add_line_2d(g, p0, p1, wp)
+    p2 = slvs.add_point_2d(g, 20, 20, wp)
+    p3 = slvs.add_point_2d(g, 0, 10, wp)
+    p4 = slvs.add_point_2d(g, 30, 20, wp)
+    slvs.distance(g, p2, p3, 40, wp)
+    slvs.distance(g, p2, p4, 40, wp)
+    slvs.distance(g, p3, p4, 70, wp)
+    slvs.distance(g, p0, p3, 35, wp)
+    slvs.distance(g, p1, p4, 70, wp)
+    line1 = slvs.add_line_2d(g, p0, p3, wp)
+    slvs.angle(g, line0, line1, 45, wp, False)
+
+    result = slvs.solve_sketch(g, 0)
+    print("result", result)
+    # self.assertEqual(result['result'], 0)
```

Comment:
Somehow the results in those tests are TOO_MANY_UNKNOWNS, although the sketch seems to have solved correctly, will have to look into what's going on here...